### DISCUSSION TITLE
MSI created in WIXToolset version 3.5, MSI Installer not able to remove the root folder during the build upgrade.

### DISCUSSION THREAD

**Discussion by logmeinndt7** [wixtoolset/issues#6883] [Questions]
When we try to upgrade from an older version to a newer version of my product, the installer is not able to remove the root folder C:\Program Files (x86)\SampleProduct. and it gets stuck and we can see the below error from MSI installation log.

> MSI (s) (64:24) [15:46:23:454]: Note: 1: 2228 2: 3: Error 4: SELECT Message FROM Error WHERE Error = 2911
> DEBUG: Error 2911: Could not remove the folder C:\Program Files (x86)\SampleProduct.
> MSI (s) (64:24) [15:46:23:456]: Executing op: ProgressTotal(Total=1,Type=1,ByteEquivalent=175000)
> The installer has encountered an unexpected error installing this package. This may indicate a problem with this package. The error code is 2911. The arguments are: C:\Program Files (x86)\SampleProduct, ,
> 
> We are expecting the MSI installer should be able to remove the C:\Program Files (x86)\SampleProduct. as it runs with high privileges but during upgrade and cleanup the root folder, it is not able to remove the root folder

**Comment by robmen** [comment]
Did you happen to remove `SYSTEM` from the ACL on that folder? I'd also check to see if you have a process holding that folder in use, preventing it from being removed.
